broker/src/main/java/org/apache/rocketmq/broker/processor/PopBufferMergeService.java

```diff
@@ -402,9 +402,14 @@ private boolean checkQueueOk(PopCheckPointWrapper pointWrapper) {
      * @param nextBeginOffset
      * @return
      */
-    public void addCkJustOffset(PopCheckPoint point, int reviveQueueId, long reviveQueueOffset, long nextBeginOffset) {
+    public boolean addCkJustOffset(PopCheckPoint point, int reviveQueueId, long reviveQueueOffset, long nextBeginOffset) {
         PopCheckPointWrapper pointWrapper = new PopCheckPointWrapper(reviveQueueId, reviveQueueOffset, point, nextBeginOffset, true);
 
+        if (this.counter.get() > brokerController.getBrokerConfig().getPopCkMaxBufferSize()) {
```

Comment:
The 

The `addCkJustOffset` will only be invoked when add ck to buffer failed.
So this condition will always be true and return false.

See addCk
https://github.com/apache/rocketmq/blob/8410785ef9d0bd2d2387bb5a706d84a4971356be/broker/src/main/java/org/apache/rocketmq/broker/processor/PopBufferMergeService.java#L447-L467

Reply:
Yeah, if the buffer full, because the limit condition, the add ck to buffer will be failed, but the `addCkJustOffset ` will add buffer success.